proof/crefine/ARM/Fastpath_C.thy

```diff
@@ -1811,9 +1811,10 @@ proof -
           apply (simp add: from_bool_0 if_1_0_0 cong: if_cong)
           apply (rule ccorres_cond_true_seq)
           apply (rule ccorres_split_throws)
-           apply (fold dc_def)[1]
            apply (rule ccorres_call_hSkip)
-             apply (rule slowpath_ccorres, simp+)
+             apply (erule disjE; simp flip: dc_def; rule slowpath_ccorres)
+            apply simp
+           apply simp
```

Comment:
what happened here? this looks like something isn't working as well as it used to

Reply:
I couldn't see what the state was before, but I suspect that the simplifier now does slightly more than before. It ended up in a disjunction `isRight .. \/ ~ isEndpoint ...`. Which is really `~ isLeft ... \/ ~ isEndpoint ..`, which probably used to be `~ (isLeft .. /\ isEndpoint)`. The simplifier can't make progress from the `.. \/ ..` point and needs an explicit `disjE`, whereas before it seems to have used the conjunction first and simplified some `IF` condition somewhere.

This proof is the only case where that happened and the fix wasn't too bad, so I didn't investigate further. It's possible that the goal was close to some simp depth limit or similar and is now over. Not clear.